00_ES6/00_start/src/pages/lobby/components/room-list.component.jsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+
+
+export const RoomListComponent = (props) =>
+  <List>
+    {props.rooms.map((room) =>
+      <ListItem button key={room.id}  onClick={(e) => props.onFieldChange('selectedRoom')(room)}>
+        <ListItemText
+          primary={room.name}
+        />
+      </ListItem>,
+    )}
+  </List>
+
+RoomListComponent.propTypes = {
+  rooms: PropTypes.array.isRequired,
+  onFieldChange : PropTypes.func.isRequired
+};
```

Comment:
Not sure if proptypes can be more detailed or concise, realized that there's a catch with Proptyes, you are adding extra wieght to your bundle (package plus all the js code, maybe on production mode is removed?) this doesn't happen on typescript ;)

Reply:
Even you can make your own custom, but I don't know if fits the issue.